### PR TITLE
Adds the fat extractor

### PR DESCRIPTION
**Pull request by Time-Green**
A machine that sucks fat from someone while giving them 'useful' tips.
At the end of the fat sucking cycle it dispenses the meat. It takes 90 nutrition for 1 slab of meat, and one slab of meat gives you about 50~ nutrition, so you can't get infinite meat.

People have 2 second to leave before the extraction starts. They can leave during the extraction cycle, unless the chef toggled the safety hatch.
The extractor does not go below starving and only starts sucking when someone is above well fed. 

Also this is the best sprite I think I've ever made 
:cl: MrDroppodBringer and Time-Green
add: Adds the lipid extractor to the service protolathe arsenal
/:cl:
Why: The kitchen can always use some love, and this allows them to turn plants into meat by using replenishable human lifestock. 

Also second PR doing this because everything shit itself

### CODE REVIEW

**Comment by adamkad1** [comment]
what it does if you emag it?

**Comment by Time-Green** [comment]
I’ll remove the nutrition restrictions on emagged, so that it’ll just starve someone to death

**Comment by adamkad1** [comment]
> I’ll remove the nutrition restrictions on emagged, so that it’ll just starve someone to death

starving to death exists?

**Comment by Time-Green** [comment]
Now that you mention it, I don’t think so. It wont be death, but a major inconvenience. That or it just tears of their flesh, but we already have a bunch of those

**Comment by adamkad1** [comment]
i know! it should turn them into skeletons!

**Comment by knacker48** [comment]
YES! Free the skeletons from their fleshy suits!

**Comment by ExcessiveUseOfCobblestone** [comment]
Honestly hoping an alternate function would be to provide small amounts of power "via burning fat".

**Comment by adamkad1** [comment]
laaame

**Comment by habader** [comment]
oh my fucking god

**Comment by PKPenguin321** [comment]
>Also emag functions are lame.

You never should have come here

**Comment by WJohn** [comment]
TBH if extracted it should instead just keep taking meat well past the point of starvation until it starts dealing hefty chunks of brute damage until the target has taken 200+ brute damage. Maybe doing something like 40 brute per slab of meat it keeps taking.

**Comment by ATH1909** [comment]
I fully support this machine turning its victim into a skeleton (if they had flesh in the first place) if they stay in it for too long after the machine has been emagged.

Maybe a skeleton with a maximum health penalty (like the one you get from the soul tap spell), though, since the skeleton race is actually pretty strong?

**Comment by Time-Green** [comment]
I really wish people would be more excited about the fat extractor, than potential emag functions.
I don't mind making it tear up flesh, but if we have it make skeletons it'll just become the skeleton making machine that everyone will abuse to become skeletons.

**Comment by WJohn** [comment]
Probably because fatness isn't something you can see or worry about often.

**Comment by Time-Green** [comment]
This is just one of those things that should've been added years ago.
it yearns to be added

**Comment by Angustmeta** [comment]
Fatness is a big deal and this gives a fun way to get rid of it.

**Comment by nicbn** [comment]
It should make you into bones (remains) but not into a skeleton when snagged.
Also it should be a medical machinery not service (most doctors are idle anyway so it gives them something interesting to work on).

**Comment by Time-Green** [comment]
It's more about the meat production than fat reduction, medbay has no use for that

**Comment by 81Denton** [comment]
I'm looking forward to botanists force feeding monkeys corn oil and then shoving them in the machine, like some sort of fucked up space foie gras farm

**Comment by GuyonBroadway** [comment]
Emag functions make for neat ways to fuck with the station that isn't just another thing that facilitates muderboning without dying. 


That being said I can't think of anything that doesn't tread on the toes of thw

**Comment by ExcessiveUseOfCobblestone** [comment]
> Emag functions make for neat ways to fuck with the station that isn't just another thing that facilitates muderboning without dying.

I don't mind the emag functions themselves persay, I hate that the only way to get these neat sabotaging features is from an exclusive item present in literally only one gamemode (albeit the most common).

**Comment by GuyonBroadway** [comment]
It's usually because its easy to call emag_act as opposed to coding in some bespoke method of hacking.

**Comment by MrDoomBringer** [comment]
image
oh god please bring this more in line with our current sprite style

**Comment by Time-Green** [comment]
i tried really hard making those sprites ok. I even matched the perspective

**Comment by Time-Green** [comment]
But yeah ok they're still terrible. What do you want me to change?

**Comment by Militaires** [comment]
to be honest with you my dude that sprite isnt that bad, like if you just fucked with the colors and contrast a little then added borders and shit it'd be fine

**Comment by Time-Green** [comment]
Added new sprites by @MrDoomBringer , also added possiblity of resisting and added more sounds

**Comment by MrDoomBringer** [comment]
image
i fucked with the colors and contrast and added borders

**Comment by MrDoomBringer** [comment]
image
new sprite^

**Comment by SouDescolado** [comment]
Can the extracted fat become an ingredient for something? Or it just poofs?

File: code/game/machinery/fat_sucker.dm [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	)
+
+/obj/machinery/fat_sucker/Initialize()
+	. = ..()
+	soundloop = new(list(src),  FALSE)
+	update_icon()
+
+/obj/machinery/fat_sucker/RefreshParts()
+	..()
+	var/rating
```
Comment (SpaceManiac) [inline code/game/machinery/fat_sucker.dm]:
```suggestion
	var/rating = 0
```

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 2]
```diff
@@ -984,3 +984,9 @@
 	name = "Pay Stand (Machine Board)"
 	build_path = /obj/machinery/paystand
 	req_components = list()
+
+/obj/item/circuitboard/machine/fat_sucker
+	name = "Fat Sucker (Machine Board)"
```
Comment (SpaceManiac) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm]: Says "Fat Sucker" here even though it's "lipid extractor" everywhere else

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
Minor stuff from a read, will check functionality later

**Comment by Time-Green** [comment]
Minor stuff fixed

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
Bug: becomes unclickable after being used once

**Comment by Time-Green** [comment]
@SpaceManiac What did you do that made it unclickable? I've tried using it a few times in a few different ways just now and I can't get the bug to show itself

**Comment by AnturK** [comment]
So what's the status on this ?

**Comment by Time-Green** [comment]
I kind of a hit a wall. @SpaceManiac  describes a bug that apparently only makes it work once, but that doesnt happen at all for me no matter what I do. 

My final guess is that he used different stock parts, so I guess I’ll try all possible tiers tomorrow.

**Comment by SpaceManiac** [comment]
Sorry, been busy. Two weeks ago so I don't remember what I did but I don't think I did anything special. I don't have time to do a detailed local test so I'll have to punt to another maintainer. Code looks good otherwise.

File: code/game/machinery/fat_sucker.dm [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+		if(!iscarbon(occupant))
+			occupant.forceMove(drop_location())
+			occupant = null
+			return
+		to_chat(occupant, "<span class='notice'>You enter [src]</span>")
+		addtimer(CALLBACK(src, .proc/start_extracting), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
+		update_icon()
+
+/obj/machinery/fat_sucker/open_machine(mob/user)
+	make_meat()
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/fat_sucker.dm]: this proc is only called when you _open_ the machine?

**Comment by ExcessiveUseOfCobblestone** [comment]
also nice trips

File: code/game/machinery/fat_sucker.dm (hunk 3 above)
Comment (Time-Green) [reply code/game/machinery/fat_sucker.dm]: Open machine also runs when it automaticly opens

**Comment by Time-Green** [comment]
I tried just building it from scraps instead of spawning it, and it works fine.

Might just be a space particle hitting your computer. yeah thats probably it

File: code/game/machinery/fat_sucker.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/fat_sucker.dm]: oh I see it pulls nutrition from the mob on close but doesn't actually make the meat until it opens.

**Pull request merged**